private/bufpkg/bufconnect/errors.go

```diff
@@ -47,3 +47,31 @@ func AsAuthError(err error) (*AuthError, bool) {
 	ok := errors.As(err, &authErr)
 	return authErr, ok
 }
+
+// AugmentedConnectError wraps an error, providing the Procedure and Addr of
+// the request for introspection in callers.
+type AugmentedConnectError struct {
+	cause     error
+	procedure string
+	addr      string
+}
+
+// Error implements the error interface and returns the error message.
+func (e *AugmentedConnectError) Error() string {
+	return e.cause.Error()
```

Comment:
Doesn't handle the nil case like `AuthError` does above, resulting in a potential panic.

Reply:
https://github.com/bufbuild/buf/pull/2871/commits/6af6dae9ddd9da097671c41b314adfa116b0cbfc